processor/src/main/java/org/mapstruct/ap/internal/model/source/SubclassMappingOptions.java

```diff
@@ -166,6 +174,41 @@ private static void addAndValidateInstance(SubclassMappingGem subclassMapping, E
         TypeMirror sourceSubclass = subclassMapping.source().getValue();
         TypeMirror targetSubclass = subclassMapping.target().getValue();
 
-        mappings.add( new SubclassMappingOptions( sourceSubclass, targetSubclass, beanMappingOptions ) );
+        mappings
+                .add(
+                    new SubclassMappingOptions(
+                        messager,
+                        typeUtils,
+                        subclassMapping,
+                        sourceSubclass,
+                        targetSubclass,
+                        beanMappingOptions ) );
+    }
+
+    public static Set<SubclassMappingOptions> copyForInverseInheritance(Set<SubclassMappingOptions> subclassMappings,
```

Comment:
Why don't we pass the `message` and `typeUtils` to this method instead of dragging them in the `SubclassMappingOptions`? It doesn't feel right to me to add those things to the model.

Reply:
I thought of doing that, but then I had trouble with determining the correct annotation etc for the message to pop-up on. I'll see if I can think of a better solution for this issue.

Reply:
Solved this by moving the SubclassValidator further outside. Now the same validator will be used for both the methods `SubclassMapping`s as well as the inverse inherited `SubclassMapping`s.